lib/core/src/Data/Quantity.hs

```diff
@@ -134,49 +136,63 @@ instance (KnownSymbol unit, Buildable a) => Buildable (Quantity unit a) where
 
 -- | Opaque Haskell type to represent values between 0 and 100 (incl).
 newtype Percentage = Percentage
-    { getPercentage :: Word }
+    { getPercentage :: Scientific }
     deriving stock (Generic, Show, Eq, Ord)
     deriving newtype (ToJSON)
 
+-- | Desired number of digits after the decimal point for presenting the
+-- @Percentage@ type.
+percentageNumberOfFractionalDigits :: Int
+percentageNumberOfFractionalDigits = 2
+
 instance NFData Percentage
 
 instance FromJSON Percentage where
-    parseJSON x = do
-        n <- parseJSON x
-        either (fail . show) return (mkPercentage @Int n)
+    parseJSON = withScientific "Percentage [0,100]" $ \s ->
+        either (fail . show) return $ mkPercentage (toRational s/100)
 
 instance Bounded Percentage where
     minBound = Percentage 0
     maxBound = Percentage 100
 
-instance Enum Percentage where
-    fromEnum (Percentage p) = fromEnum p
-    toEnum = either (error . ("toEnum: " <>) . show) id . mkPercentage
-
 instance ToText Percentage where
-    toText (Percentage p) = T.pack (show p) <> "%"
+    toText (Percentage p) = T.pack (showS p) <> "%"
+      where
+        showS = formatScientific
+            Fixed
+            (Just percentageNumberOfFractionalDigits)
```

Comment:
@KtorZ 
> Why not choose Rational for the inner type of the Percentage type and do the formatting / cast to 2-decimal scientific only at the API boundaries

My thinking was that it would be safer to concentrate the rounding to the only constructor of `Percentage` (I removed the `Enum` instance also). I.e not have to remember to round in each `ToText`, `ToJSON` instance, etc.

But I guess I already use `formatScientific`, and cold as just as well move the `round' :: Scientific -> Scientific` function from `mkPercentage` to `toJSON`…

Reply:
Or… do we _really_ have to?

Reply:
Done.